### DISCUSSION TITLE
[Bug]: Bing problem

### DISCUSSION THREAD

**Discussion by Maddaxman** [danny-avila/LibreChat#633] [Troubleshooting] [answered]
### What happened?

It seems that Bing has upgraded its strategy, which has caused it to malfunction now. Is there a way to solve this?

<img width="918" alt="image" src="https://github.com/danny-avila/LibreChat/assets/31379908/6652607f-1da0-432f-93fa-479949e6c773">

<img width="1089" alt="image" src="https://github.com/danny-avila/LibreChat/assets/31379908/deb3857b-0e3f-45c3-b807-48c59530cd4f">

### Steps to Reproduce

I have tried to change the cookie. Previously, it would expire after a while, but now even if I change it, it still cannot be used.

And I can use Sydney normally in ChatGPTBox.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Bing is pretty fickle. Have you tried:
- without a cookie
- full cookie string
- just the _U cookie string? 
    - example:  `_U=your_U_cookie_string_here`

**Comment by fuegovic** [comment, marked as answer]
Are you using a VPN, or an instance deployed on an hosting service (as opposed to running LibreChat locally on your computer) ?

This error message is the one we get when the ip has been blocked and we use the full cookie string. It doesn't usually happen when running locally but I've seen that exact error message when I was hosting from a provider that had the IP blocked.

You can always try to refresh your cookie (provide a new `full cookie string`)

If you completly remove your token, do you get the same error or the ip blocked message?

I'm affraid there's not much to do if the issue is the ip being blocked. Microsoft is actively banning IPs with too much traffic from accessing bing chat to prevent abuse.

Let me know how it goes!

**Comment by Nicolas-One** [comment]
I have been facing the same issue for a while now, unable to use Bing. I am deploying on Oracle Cloud, and even after changing my IP address, I still cannot access it. I have tried using complete cookies or using "_U=xxxx," but I keep getting the same error. It's frustrating that I cannot use it anymore.

**Reply by fuegovic** [reply]
Sadly there's nothing we can do about it. Microsoft is blocking IPs from VPNs and cloud computing providers. You must understand that Microsoft do not endorse the way we use Bing. We are using workaround to bypass the limitations. They are blocking the IPs of these services providers to prevent abuse. What they really want is more Edge users and more traffic on bing.com, that's the only reason they are offering this service for free.

I self host my instance of LibreChat from an old pc at home to be able to access it from anywhere. As long as I have a reasonable ammount of users (a couple of friends and familly members) I wont have any issue with bing being blocked. It might not be ideal but it's the only way to access bing remotely with LibreChat at the moment.

**Comment by danny-avila** [comment]
@Maddaxman looks like it might be related to this issue: https://github.com/waylaidwanderer/node-chatgpt-api/issues/437

It was fixed in the dependency, so I will update it and let us know if you still experience. Will report back when I update it
